test/savecontext-ut.cc

```diff
@@ -0,0 +1,54 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present
+// Datadog, Inc.
+
+#include <gtest/gtest.h>
+
+#include "perf.h"
+#include "savecontext.hpp"
+#include "unwind.hpp"
+#include "unwind_state.hpp"
+
+#include <algorithm>
+#include <unistd.h>
+
+void funcA() __attribute__((noinline));
+void funcB() __attribute__((noinline));
+
+std::byte stack[PERF_SAMPLE_STACK_SIZE];
+
+void funcB() {
+  UnwindState state;
+  uint64_t regs[K_NB_REGS_UNWIND];
+  size_t stack_size = save_context(regs, stack);
```

Comment:
I thought span construction was explicit. TIL

Reply:
Constructors from C arrays / std::array are not explicit, which is nice !